### DISCUSSION TITLE
v0.14

### DISCUSSION THREAD

**Discussion by Louisvdw** [Louisvdw/dbus-serialbattery#315] [Announcements]
## What's Changed

NB! Sinowealth users - The Sinowealth driver has been disabled because of interference. You need to remove the comment # on line 21 in utils.py before the driver will work for Sinowelath BMS.

* Implemented constants for charge and dischare limits by @corneel27 in https://github.com/Louisvdw/dbus-serialbattery/pull/238
* Esc LiPro cells by @Louisvdw in https://github.com/Louisvdw/dbus-serialbattery/pull/251
* Constants for soc low by @corneel27 in https://github.com/Louisvdw/dbus-serialbattery/pull/267
* moved battery_types to utils.py by @corneel27 in https://github.com/Louisvdw/dbus-serialbattery/pull/270
* Implemented CCL/DCL depending on cell-voltage by @WaldemarFech in https://github.com/Louisvdw/dbus-serialbattery/pull/293
* Implemented CVL depending on cell-voltage by @WaldemarFech in https://github.com/Louisvdw/dbus-serialbattery/pull/307
* Settings for UI by @Louisvdw in https://github.com/Louisvdw/dbus-serialbattery/pull/252
* Linear limits by @Louisvdw in https://github.com/Louisvdw/dbus-serialbattery/pull/312
* v0.14 by @Louisvdw in https://github.com/Louisvdw/dbus-serialbattery/pull/314

## New Contributors
* @corneel27 made their first contribution in https://github.com/Louisvdw/dbus-serialbattery/pull/238
* @WaldemarFech made their first contribution in https://github.com/Louisvdw/dbus-serialbattery/pull/293

**Full Changelog**: https://github.com/Louisvdw/dbus-serialbattery/compare/v0.13...v0.14

<hr /><em>This discussion was created from the release <a href='https://github.com/Louisvdw/dbus-serialbattery/releases/tag/v0.14'>v0.14</a>.</em>

**Comment by coca7** [comment]
I like the formatting in the new version of the driver explaining the features.

**Comment by DirkReudenbach** [comment]
Thank you for the new and very good further developments but sorry - which of the Limitation modes is standard and how to switch between the Limitation Mode?

**Reply by Louisvdw** [reply]
Step is the standard limiting mode. The other option is Linear.
For both these options you can use any of the limiting attributes (SOC, Cell Voltage, Temperature)

By default the mode is Step with all 3 attributes enabled for Charge and Discharge.
The old "classic" way is the same as Step with SOC for for Charge and Discharge.
Waldemar's version would be the same as Linear with Cell Voltage & Temprature for Charge and Discharge

https://github.com/Louisvdw/dbus-serialbattery/wiki/Features#charge-current-control-management

**Reply by DirkReudenbach** [reply]
ok thanks for your quick response and got it :-) and maybe you can help me on how can I switch to linear mode?
If I change the # in front of the Limitation modes from Step to Linear than the following happens like listed in issue 317 
https://github.com/Louisvdw/dbus-serialbattery/issues/317

**Comment by merlep** [comment]
I see Total AH drawn is listed as available - in Feature table its listed as "Calc" for the JK BMS - what does that mean ? Is it calculated by dbus from first instance or does it rely on the actual BMS to provide the value ?
thanks
